### PR TITLE
updated np.product to np.prod

### LINKED ISSUE #2248: np.product is deprecated in numpy v1.25
**Describe the bug**
np.product is deprecated in v1.25.
https://github.com/numpy/numpy/releases/tag/v1.25.0rc1

There are a few instances currently in arviz. See [here](https://github.com/search?q=repo%3Aarviz-devs%2Farviz%20np.product&type=code)

**To Reproduce**
N/A

**Expected behavior**
np.product can be replaced by np.prod

### PR DESCRIPTION
**Pull request by Surya-Narayan**
Fix #2248 
np.product is deprecated from numpy v1.25 and above. Replaced np.product to np.prod

## Checklist

- [X] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format
- [X] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/main/CHANGELOG.md#v0xx-unreleased)

----
:books: Documentation preview :books:: https://arviz--2249.org.readthedocs.build/en/2249/

### CODE REVIEW

**Review by nilanjan2002 — APPROVED** [review]
Good Work!

**Comment by Surya-Narayan** [comment]
@ahartikainen Is there any issue with the changes I have made?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 - InferenceData objects can now be appended to existing netCDF4 files and to specific groups within them ([2227](https://github.com/arviz-devs/arviz/pull/2227))
 
 ### Maintenance and fixes
+- Fix np.product issue by replacing with np.prod ([2248](https://github.com/arviz-devs/arviz/pull/2249))
```
Comment (OriolAbril) [inline CHANGELOG.md:11]:
```suggestion
- Replace deprecated np.product with np.prod ([2249](https://github.com/arviz-devs/arviz/pull/2249))
```

**Comment by OriolAbril** [comment]
Thanks @Surya-Narayan !

**Comment by Surya-Narayan** [comment]
Thank you :)

**Pull request merged**